### PR TITLE
Update documentation for generateFeatures task

### PR DESCRIPTION
**Pull request by kathrynkodama**
See https://github.com/OpenLiberty/ci.maven/pull/1522 for the same documentation updates in ci.maven

Signed-off-by: Kathryn Kodama <kathryn.s.kodama@gmail.com>

### CODE REVIEW

**Review by turkeylurkey — APPROVED** [review]
I'd like to suggest not reformatting too much because it can make the differences hard to find.

**Review by TrevCraw — APPROVED** [review]

File: docs/generateFeatures.md [hunk 1]
```diff
@@ -1,58 +1,52 @@
...
+If this task detects Liberty features used in your project but not present in your Liberty configuration, it will create a new file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` directory of your project. The `generated-features.xml` file will contain a list of features required for your project. If the `generated-features.xml` file has been created in the past and no additional features have been detected, this file will be retained.
 
-The task examines the `build.gradle` dependencies to determine what version of Java EE and what version of MicroProfile you may be using. It will then generate features which are compatible. For Java EE the task looks for these coordinates:
-* javax:javaee-api:6.0
-* javax:javaee-api:7.0
-* javax:javaee-api:8.0
-* jakarta:jakartaee-api:8.0
+If you are using [devc](libertyDev.md#libertydevc-task-container-mode), ensure that the `generated-features.xml` configuration file is copied via your Dockerfile.
```
Comment (TrevCraw) [inline docs/generateFeatures.md:15]:
```suggestion
If you are using [devc](libertyDev.md#libertydevc-task-container-mode), ensure that the `generated-features.xml` configuration file is copied to your Docker image via your Dockerfile.
```

File: docs/libertyDev.md [hunk 2]
```diff
@@ -109,6 +112,11 @@ When dev mode runs with container support, it builds a container image and runs
 
 **Limitations:** This task is not supported with Spring Boot applications.
 
+N.B. starting in 3.4, dev mode invokes `generate-features`. Ensure that the `generated-features.xml` configuration file is copied via your Dockerfile.
```
Comment (TrevCraw) [inline docs/libertyDev.md:115]:
```suggestion
N.B. starting in 3.4, dev mode invokes `generate-features`. Ensure that the `generated-features.xml` configuration file is copied to your Docker image via your Dockerfile.
```

File: docs/generateFeatures.md [hunk 3]
```diff
@@ -1,58 +1,52 @@
...
-        mavenCentral()
-        maven {
-            url "https://plugins.gradle.org/m2"
-            url "https://oss.sonatype.org/content/repositories/snapshots"
-        }
-    }
-    dependencies {
-        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.3.1-SNAPSHOT'
-    }
+For example, if you have the following Jakarta EE and MicroProfile dependencies in your `build.gradle` file, features compatible with Jakarta EE `8.0` and MicroProfile `4.1` will be generated.
```
Comment (TrevCraw) [inline docs/generateFeatures.md:28]:
```suggestion
For example, if you have the following Jakarta EE and MicroProfile dependencies in your `build.gradle` file, features compatible with Jakarta EE `8.0.0` and MicroProfile `4.1` will be generated.
```

File: docs/generateFeatures.md [hunk 4]
```diff
@@ -1,58 +1,52 @@
...
+If you are using [devc](libertyDev.md#libertydevc-task-container-mode), ensure that the `generated-features.xml` configuration file is copied via your Dockerfile.
+```dockerfile
+COPY --chown=1001:0  build/wlp/usr/servers/defaultServer/configDropins/overrides/generated-features.xml /config/configDropins/overrides/
+```
 
-For MicroProfile it looks for `org.eclipse.microprofile:microprofile` and generates features according to the version number. The task uses these compile dependencies to determine the best Liberty features to use with your application. 
+The task examines the `build.gradle` dependencies to determine which version of Jakarta EE, MicroProfile or Java EE API you may be using. Compatible features will then be generated.
 
-The task also considers the features you have already specified in `server.xml` or other files that Liberty will use (e.g. `include` elements and `configDropins` files). The task will attempt to find a working set of features that are compatible with each other.
+For Jakarta EE API, this task looks for `jakarta:jakartaee-api` dependency with version `8.0`.
```
Comment (TrevCraw) [inline docs/generateFeatures.md:22]:
```suggestion
For Jakarta EE API, this task looks for `jakarta:jakartaee-api` dependency with version `8.0.0`.
```

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: docs/generateFeatures.md [hunk 5]
```diff
@@ -12,6 +12,11 @@ This task is not part of the lifecycle, so to use it in your build you will need
 
 If this task detects Liberty features used in your project but not present in your Liberty configuration, it will create a new file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` directory of your project. The `generated-features.xml` file will contain a list of features required for your project. If the `generated-features.xml` file has been created in the past and no additional features have been detected, this file will be retained.
 
+If you are using [devc](libertyDev.md#libertydevc-task-container-mode), ensure that the `generated-features.xml` configuration file is copied via your Dockerfile.
+```dockerfile
+COPY --chown=1001:0  build/wlp/usr/servers/defaultServer/configDropins/overrides/generated-features.xml /config/configDropins/overrides/
```
Comment (turkeylurkey) [inline docs/generateFeatures.md:17]: Suggestion: another sample COPY for Linux that copies the whole `overrides` dir.

**Review by turkeylurkey — APPROVED** [review]

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**